ios/Codeforces Watcher/Application/features/contests/ContestsView.swift

```diff
@@ -0,0 +1,107 @@
+                        NoItemsView(imageName: "noItemsContests", text: noContestsExplanation)
+                    } else {
+                        ContestsList
+                    }
+                }, refreshControl: refreshControl)
+                    .background(Palette.white.swiftUIColor)
+                    .cornerRadius(30, corners: [.topLeft, .topRight])
+            }
+        }
+        .background(Palette.accentGrayish.swiftUIColor.edgesIgnoringSafeArea(.top))
+    }
+    
+    @ViewBuilder
+    private var ContestsList: some View {
+        ScrollView {
+            LazyVStack {
+                ForEach(contests.indices, id: \.self) { index in
+                    let contest = contests[index]
+                    
+                    if index == 0 || contest.month != contests[index - 1].month {
+                        CommonText(contest.month.capitalizingFirstLetter())
+                            .font(.midHeaderSemibold2)
+                            .foregroundColor(Palette.black.swiftUIColor)
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                            .padding(.top, 15)
+                            .padding(.bottom, 5)
+                            .padding(.horizontal, 20)
+                    }
+                    
+                    ContestView(contest.contest)
+                        .contentShape(Rectangle())
+                        .onTapGesture {
+                            self.onContest(contest.contest)
+                        }
+                }
+            }
+        }
+    }
+    
+    private func ContestView(_ contest: Contest) -> some View {
+        let name = contest.title
+        let logoName = Contest.Platform.getImageNameByPlatform(contest.platform)
+        let date = Double(contest.startDateInMillis / 1000).secondsToContestDateString()
+        
+        return HStack(spacing: 8) {
+            Image(logoName)
+                .resizable()
+                .frame(width: 36, height: 36)
+                .clipShape(Circle())
+            
+            VStack(spacing: 4) {
+                CommonText(name)
+                    .font(.bodySemibold)
+                    .foregroundColor(Palette.black.swiftUIColor)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                
+                CommonText(date)
+                    .font(.hintRegular)
+                    .foregroundColor(Palette.darkGray.swiftUIColor)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+            }
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .lineLimit(1)
+            
+            Button(action: {
+                onCalendar(contest)
+            }, label: {
+                Image("calendarAddIcon")
+                    .frame(width: 18, height: 20)
+            })
+        }
+        .padding(.horizontal, 20)
+        .padding(.vertical, 10)
+    }
+    
+    struct UIModel {
```

Comment:
In `UiModel` you should have only the data you use in the view, so passing `Contest` is a bit overkill.

Reply:
Contests is passed for `onCalendar` and `onContest` functions, which use the required fields.
Also, I can create a struct with required fields for the view and this functions and pass less data as a result.